bin/dfx-software-dfx-install

```diff
@@ -14,8 +14,14 @@ source "$(clap.build)"
 [[ "${DFX_VERSION}" != "latest" ]] || DFX_VERSION=""
 DFX_VERSION="${DFX_VERSION#v}"
 
+# Install the dfx cache if it hasn't been installed already
+lazy_dfx_cache_install() {
```

Comment:
What do you mean by lazy?
In my mind it would be lazy if you only install it when you need it (like you do in `bin/dfx-sns-wasm-upload`).
Here it looks proactive to me.
But maybe you're referring to something else?

Reply:
I mean that it doesn't do unnecessary work installing the cache if it is already installed.  For sure it doesn't correspond to the functional programmer's definition of lazy but it matches the Oxford definition. :-)

Reply:
For contrast, `dfx cache install` will reinstall the cache, even if it is already there.